### PR TITLE
Manage many products, many distribution of product, and the repo state

### PR DESCRIPTION
**Pull request by cdelgehier**
I tried to simplify the role:
- I removed the Debian parts (this linux distribution is not supported atm)
- I've harmonized RedHat & Ubuntu tasks
- I added some Booleans to be a little more polite with the user
- I added a state to the presence of the repo
- I made it possible to have several distribution's repo of installed products
- I added the continuous integration test

### CODE REVIEW

File: defaults/main.yml
Comment (vincent-legoll) [inline defaults/main.yml:4]: This is not the (linux) distribution's version number but the openio release number, I think the naming was better as "_release"

File: meta/main.yml
Comment (vincent-legoll) [inline meta/main.yml]: Why remove the email ?

File: meta/main.yml
Comment (vincent-legoll) [inline meta/main.yml]: That's not a company, nor a web site URL, I'll keep it at "OpenIO"

File: tasks/Debian.yml
Comment (vincent-legoll) [inline tasks/Debian.yml:2]: You removed the backports handling, are you sure this is not needed any more ?

File: README.md
Comment (vincent-legoll) [inline README.md]: "specific linux distribution" ?

File: README.md
Comment (vincent-legoll) [inline README.md:20]: I'd prefer "release", distribution sounds too much like "linux distribution"

File: README.md
Comment (vincent-legoll) [inline README.md]: You could have put the openstack handling in a separate commit, but that's more for reviewing & merging's sake.

File: README.md
Comment (vincent-legoll) [inline README.md]: maybe "password" will be more explicit (I know it was my naming, but maybe it was a wrong choice after all)

File: vars/RedHat.yml
Comment (vincent-legoll) [inline vars/RedHat.yml]: why keep that ?

File: tasks/Debian-present.yml
Comment (vincent-legoll) [inline tasks/Debian-present.yml]: OpenStack (multiple occurences)

File: defaults/main.yml
Comment (vincent-legoll) [inline defaults/main.yml:15]: This looks like deb-specific, maybe this should show in the variable name to avoid confusion.
what about `openio_repository_deb_disable_policy_autostart`, and if we have redhat specific ones we can use `openio_repository_rpm_` prefix...

File: README.md
Comment (vincent-legoll) [inline README.md]: not only openio's, but also openstack...

File: README.md
Comment (vincent-legoll) [inline README.md]: Why not give a realistic example like "oiofs" or "replicator" instead ?

File: README.md
Comment (vincent-legoll) [inline README.md]: Maybe add a comment telling that this password will be provided to customers by OpenIO.

File: README.md
Comment (vincent-legoll) [inline README.md]: trailing slash

**Review by vincent-legoll — CHANGES_REQUESTED** [review]
We'll also want to support debian distros alongside the ubuntus. But that support may be postponed.

File: defaults/main.yml
Comment (cdelgehier) [reply defaults/main.yml:4]: A `distribution` is a software **collection**. The binaries in this distribution can be present in various version/release.

File: meta/main.yml
Comment (cdelgehier) [reply meta/main.yml]: There is no email in the skeleton https://github.com/open-io/ansible-role-openio-skeleton/blob/master/meta/main.yml#L2

File: meta/main.yml
Comment (cdelgehier) [reply meta/main.yml]: bitmoji

File: tasks/Debian.yml
Comment (cdelgehier) [reply tasks/Debian.yml:2]: https://github.com/open-io/ansible-role-openio-repository/blob/master/tasks/Debian.yml#L7 Debian is not supported atm

File: README.md
Comment (cdelgehier) [reply README.md]: nope specific distribution of the product :) :vulgarize:

File: README.md
Comment (cdelgehier) [reply README.md:20]: ditto

File: README.md
Comment (cdelgehier) [reply README.md]: yep, next time

File: README.md
Comment (cdelgehier) [reply README.md]: ok password is more precise :) Just like `distribution`

File: vars/RedHat.yml
Comment (cdelgehier) [reply vars/RedHat.yml]: delete

File: tasks/Debian-present.yml
Comment (cdelgehier) [reply tasks/Debian-present.yml]: replaced

File: defaults/main.yml
Comment (cdelgehier) [reply defaults/main.yml:15]: The long names does not bother me, but a description in the README seems sufficient

File: README.md
Comment (cdelgehier) [reply README.md]: fixed

File: README.md
Comment (cdelgehier) [reply README.md]: right, fixed

File: README.md
Comment (cdelgehier) [reply README.md]: done in description

File: README.md
Comment (cdelgehier) [reply README.md]: done

File: defaults/main.yml
Comment (vincent-legoll) [reply defaults/main.yml:15]: OK, so what about putting it in a `defaults/Debian.yml` file instead ? In that case you can keep the name without the "_deb_"...

File: meta/main.yml
Comment (vincent-legoll) [reply meta/main.yml]: OK, so now the question becomes why is the email not in the skeleton ? I always see emails alongside names wherever I see developer or maintainer's names... No need to look them up if you want to contact them...

 I won't be fighting for that though, that was just a genuine question.

Comment (cdelgehier) [reply meta/main.yml]: I approuve ! So make a PR on the skeleton :)
I can add the email

Comment (cdelgehier) [reply meta/main.yml]: done

File: defaults/main.yml
Comment (cdelgehier) [reply defaults/main.yml:15]: it s works ? `defaults/Debian.yml`  I've never seen this before

Comment (vincent-legoll) [reply defaults/main.yml:15]: It looks like ansible will never support that, c.f.:
https://github.com/ansible/ansible/issues/6756

But you can do some manual processing, like :
https://github.com/sbitio/ansible-fail2ban/commit/70a6b594a3d8e6789119751f675cfd593b6f2a3e

I'll let you decide if you want to go down that path.

File: meta/main.yml
Comment (racciari) [inline meta/main.yml:5]: The company name is OpenIO or OpenIO SAS at the moment but not openio.io

File: README.md
Comment (racciari) [reply README.md:20]: Yep, should be release here.

File: README.md
Comment (racciari) [inline README.md:22]: Same here, Pike is the release version.

File: README.md
Comment (racciari) [inline README.md:43]: You must be thinking 17.04 here.

File: tasks/Debian.yml
Comment (racciari) [reply tasks/Debian.yml:2]: Please, let the Debian support if it's not harmful.

**Pull request merged**
